### PR TITLE
[Workshop] - open closed requirement to workshop #1485

### PR DESCRIPTION
**Pull request by Pabloostr**
#1485

### CODE REVIEW

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 1]
```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
...
+    const dialogRef = this.dialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: this.workshopData.status === 'Open' ? ModalConfirmationType.closeSet : ModalConfirmationType.openSet,
+      },
+    });
+    dialogRef.afterClosed().subscribe((result: boolean)  => {
+      if(result){
+        if(this.workshopData.status === 'Open') {
+          this.workshopData.status = 'Closed';
```
Comment (litvinets) [inline src/app/shared/components/workshop-card/workshop-card.component.ts:123]: create please enum for workshop status

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 2]
```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
...
+        }else if (this.workshopData.status === 'Closed'){
+          this.workshopData.status = 'Open';
+        } 
+        this.store.dispatch(new UpdateStatus({"workshopId": this.workshopData.workshopId, "status": this.workshopData.status}));
```
Comment (litvinets) [inline src/app/shared/components/workshop-card/workshop-card.component.ts:127]: redundant "" 
image

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 3]
```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
...
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: this.workshopData.status === 'Open' ? ModalConfirmationType.closeSet : ModalConfirmationType.openSet,
+      },
+    });
+    dialogRef.afterClosed().subscribe((result: boolean)  => {
+      if(result){
+        if(this.workshopData.status === 'Open') {
+          this.workshopData.status = 'Closed';
+        }else if (this.workshopData.status === 'Closed'){
```
Comment (litvinets) [inline src/app/shared/components/workshop-card/workshop-card.component.ts:124]: it is better to create a const Open/Close and assign it to the object afterwords

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 4]
```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
     this.isFavorite = !this.isFavorite;
   }
 
+  onChange(): void {
```
Comment (litvinets) [inline src/app/shared/components/workshop-card/workshop-card.component.ts:113]: bad naming, it should be more specific, for example on onChangeWorkshopStatus

File: src/app/shared/components/workshop-card/workshop-card.component.ts [hunk 5]
```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
...
+    const dialogRef = this.dialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: this.workshopData.status === 'Open' ? ModalConfirmationType.closeSet : ModalConfirmationType.openSet,
```
Comment (litvinets) [inline src/app/shared/components/workshop-card/workshop-card.component.ts:117]: is see you compare with status two times, probably it is better to create a common private method that will accept status and modal window type (this method will open/close modal window and then dispatch change status with proper status)
and pass a needed parameter from the outside, even from the template there can be two different buttons that are displayed according to the condition (it's up to you) 
image

File: src/app/shared/models/workshop.model.ts [hunk 6]
```diff
@@ -121,8 +121,6 @@ export interface WorkshopCard {
   workshopId: string;
   coverImageId?: string;
   directionsId: number[];
-  availableSeats: number;
-  takenSeats: number;
```
Comment (litvinets) [inline src/app/shared/models/workshop.model.ts:125]: why?

File: src/app/shared/services/workshops/user-workshop/user-workshop.service.ts [hunk 7]
```diff
@@ -73,6 +73,14 @@ export class UserWorkshopService {
     return this.http.put('/api/v2/Workshop/Update', formData);
   }
 
+  /**
+   * This method update workshop status
+   * @param workshop: Workshop
+   */
+  updateWorkshopStatus(workshop: Workshop): Observable<object> {
+    return this.http.put('/api/v1/Workshop/UpdateStatus', workshop);
```
Comment (litvinets) [inline src/app/shared/services/workshops/user-workshop/user-workshop.service.ts:81]: payload is not really correct (it should be an object with id and status only
image

File: src/app/shared/store/user.state.ts [hunk 8]
```diff
@@ -776,6 +772,14 @@ export class UserState {
     dispatch(new ShowMessageBar({ message: 'На жаль виникла помилка', type: 'error' }));
   }
 
+  @Action(UpdateStatus)
+  updateStatus({ dispatch }: StateContext<UserStateModel>, { payload }: UpdateWorkshop): Observable<object> {
+    return this.userWorkshopService.updateWorkshopStatus(payload).pipe(
+      tap(res => dispatch(new OnUpdateStatusSuccess(res))),
```
Comment (litvinets) [inline src/app/shared/store/user.state.ts:778]: typo

**Review by litvinets — CHANGES_REQUESTED** [review]
PR build is broken

File: src/app/shared/components/workshop-card/workshop-card.component.ts (hunk 2 above)
Comment (Pabloostr) [reply src/app/shared/components/workshop-card/workshop-card.component.ts:127]: done

File: src/app/shared/components/workshop-card/workshop-card.component.ts (hunk 4 above)
Comment (Pabloostr) [reply src/app/shared/components/workshop-card/workshop-card.component.ts:113]: done

File: src/app/shared/models/workshop.model.ts (hunk 6 above)
Comment (Pabloostr) [reply src/app/shared/models/workshop.model.ts:125]: done

File: src/app/shared/components/workshop-card/workshop-card.component.ts (hunk 1 above)
Comment (Pabloostr) [reply src/app/shared/components/workshop-card/workshop-card.component.ts:123]: done

File: src/app/shared/store/user.actions.ts [hunk 9]
```diff
@@ -222,6 +222,18 @@ export class OnUpdateWorkshopSuccess {
...
+  static readonly type = '[user] update Status';
+  constructor(public payload) { }
+}
+export class OnUpdateStatusSuccess {
+  static readonly type = '[user] update Status success';
+  constructor(public payload) { }
+}
+export class OnUpdateStatusFail {
+  static readonly type = '[user] update Status fail';
+  constructor(public payload: HttpErrorResponse) { }
```
[the remark this answers is not in the document]
Comment (Pabloostr) [reply src/app/shared/store/user.actions.ts:235]: done

**Pull request closed without merging**